src/vaillant_netatmo_api/thermostat.py

```diff
@@ -93,6 +95,65 @@ async def async_get_thermostats_data(self) -> list[Device]:
 
         return [Device(**device) for device in body["body"]["devices"]]
 
+    async def async_get_dhw_data(
+        self,
+        device_id: str,
+        home_id: str,
+    ) -> None:
+        """
+        Get DHW data from the Netatmo API.
+
+        """
+
+        path = _GET_DHW_DATA_PATH
+        data = {
+            "device_id": device_id,
+            "home_id": home_id,
+        }
+
+        body = await self._post(
+            path,
+            data=data,
+        )
+
+        if body["status"] != _RESPONSE_STATUS_OK:
+            raise NonOkResponseException("Unknown response error. Check the log for more details.", path=path, data=data, body=body)
+
+        return body["body"]["home"]["modules"][0]["dhw_always_on"]
+
+    async def async_set_dhw_data(
+        self,
+        device_id: str,
+        home_id: str,
+        dhw_always_on: bool,
+    ) -> None:
+        """
+        Set DHW data as json post to the Netatmo API.
+
+        """
+
+        path = _SET_DHW_DATA_PATH
+        data = {
+            "home": {
+                "modules":[
+                    {
+                        "dhw_always_on":dhw_always_on,
+                        "id":device_id
+                    }
+                ],
+                "id": home_id
+            }
+        }
+
+        body = await self._json_post(
+            path,
+            data=data,
+        )
```

Comment:
Here is the PR I mentioned: https://github.com/MislavMandaric/vaillant-netatmo-api/pull/104

It's missing some tests for the new functionality, but it didn't break any of the old ones 😅 I'll try to write some tests today, but if I don't, I'll just merge it and leave it unreleased, but on master. In that case, you can rebase your changes and have this code ready to be used in your branch.

Reply:
Well, maybe it is implemented, but I did not find the way how to set proper names of params. i.e.:
{"home": {"modules": [{"dhw_always_on": false, "id": "gw_hw_addr"}],
"id": "my_home_id"
} }
so post form data params should be:
home_id=my_home_id
access_token=my_token
device_id=gw_hw_addr
but I did not figure it how i post dhw_always_on param.

So maybe you will have some ideas.

Reply:
BTW {"home": {"`modules`": [{"dhw_always_on": false, "id": "`gw_hw_addr`"}],"id": "my_home_id"} }

gw_hw_addr is the addr of gateway (device) not the thermostat addr (module). 
Their api is not very intuitive.

Reply:
I tried it out as well and couldn't manage to make it work.

Let's keep this call with JSON request data, as you've implemented it, because we know that one works 😄 I just merged the PR I mentioned. With that work done, this is how you can now implement it:

```
body = await self._post(
            path,
            json=data,
        )
```

Since the `_post_json` method at that point will not be used anymore, you can delete your changes in `base.py` from the PR.